raster_api/runtime/src/alternate_reader.py

```diff
@@ -0,0 +1,96 @@
+"""Custom Alternate Asset Reader."""
+
+from typing import Any, Dict, Optional, Set, Type
+
+import attr
+import pystac
+import rasterio
+from morecantile import TileMatrixSet
+from rasterio.crs import CRS
+from rio_tiler.constants import WEB_MERCATOR_TMS, WGS84_CRS
+from rio_tiler.errors import InvalidAssetName, MissingAssets
+from rio_tiler.io import BaseReader, MultiBaseReader, Reader
+from rio_tiler.io.stac import DEFAULT_VALID_TYPE, _get_assets
+from rio_tiler.types import AssetInfo
+
+
+@attr.s
+class PgSTACReaderAlt(MultiBaseReader):
+    """Custom STAC Reader for the alternate asset format used widely by NASA.
+
+    Only accept `pystac.Item` as input (while rio_tiler.io.STACReader accepts url or pystac.Item)
+
+    """
+
+    input: pystac.Item = attr.ib()
+
+    tms: TileMatrixSet = attr.ib(default=WEB_MERCATOR_TMS)
+    minzoom: int = attr.ib()
+    maxzoom: int = attr.ib()
+
+    geographic_crs: CRS = attr.ib(default=WGS84_CRS)
+
+    include_assets: Optional[Set[str]] = attr.ib(default=None)
+    exclude_assets: Optional[Set[str]] = attr.ib(default=None)
+
+    include_asset_types: Set[str] = attr.ib(default=DEFAULT_VALID_TYPE)
+    exclude_asset_types: Optional[Set[str]] = attr.ib(default=None)
+
+    reader: Type[BaseReader] = attr.ib(default=Reader)
+    reader_options: Dict = attr.ib(factory=dict)
+
+    ctx: Any = attr.ib(default=rasterio.Env)
+
+    def __attrs_post_init__(self):
+        """Fetch STAC Item and get list of valid assets."""
+        self.bounds = self.input.bbox
+        self.crs = WGS84_CRS
+
+        self.assets = list(
+            _get_assets(
+                self.input,
+                include=self.include_assets,
+                exclude=self.exclude_assets,
+                include_asset_types=self.include_asset_types,
+                exclude_asset_types=self.exclude_asset_types,
+            )
+        )
+        if not self.assets:
+            raise MissingAssets("No valid asset found")
+
+    def _get_asset_info(self, asset: str) -> AssetInfo:
+        """Validate asset names and return asset's url.
+        Args:
+            asset (str): STAC asset name.
+        Returns:
+            str: STAC asset href.
+        """
+        if asset not in self.assets:
+            raise InvalidAssetName(f"{asset} is not valid")
+
+        asset_info = self.input.assets[asset]
+        extras = asset_info.extra_fields
+
+        if ("alternate" not in extras) or ("s3" not in extras["alternate"]):
+            raise MissingAssets("No alternate asset found")
+
+        info = AssetInfo(url=extras["alternate"]["s3"]["href"], metadata=extras)
+
+        if "file:header_size" in asset_info.extra_fields:
+            h = asset_info.extra_fields["file:header_size"]
+            info["env"] = {"GDAL_INGESTED_BYTES_AT_OPEN": h}
+
+        if requester_pays := extras["alternate"]["s3"].get("storage:requester_pays"):
+            if requester_pays:
+                info["env"] = {"AWS_REQUEST_PAYER": "requester"}
```

Comment:
maybe there is no `file` extension but just in case I think we should not overwrite the `info["env"]` completely 

```python
info["env"] = {}

if "file:header_size" in asset_info.extra_fields:
    h = asset_info.extra_fields["file:header_size"]
    info["env"].update({"GDAL_INGESTED_BYTES_AT_OPEN": h})

if requester_pays := extras["alternate"]["s3"].get("storage:requester_pays"):
    if requester_pays:
        info["env"].update({"AWS_REQUEST_PAYER": "requester"})
```